ui/src/models/mobileapps/mobileapp.js

```diff
@@ -43,18 +38,11 @@ export class MobileApp {
   }
 
   setProperty(propertyName, propertyValue) {
-    switch (propertyName) {
-      default:
-        this.getSpec().set(propertyName, propertyValue);
-        break;
-    }
+    this.getSpec().set(propertyName, propertyValue);
```

Comment:
This way we don't abstract the JSON structure anymore. I mean: when we have to set the name of the application, I would expect to be able to do `model.setProperty('newName')`, while with this code I have to do `model.setProperty('spec.name', 'newName')` and, in my opinion, we lose the usefulness of the model classes.
The idea that was behind the `switch/case` was that we could add there all the property whose path we want to abstract.

Reply:
I am not sure I follow you here. The method is the same as before, just the implementation a bit different

Reply:
I put the comment here in the `setProperty`, but it applies to the `getProperty` as well. Suppose you want to get the version of the app. With the current implementation, we can't use the `getPropery` (nor the `setProperty`), but we have to get the `Status` object and get the version from there.
The idea behind the `switch/case` was that you could do `getProperty('version')` and we just had to add a 
```js
case 'version':
  return this.getStatus().getVersion();
```

Reply:
ok, if this is the case, then should we not move all the methods in `AppStatus` and `AppSpec` to `MobileApp`, and not exposing the `status` or `spec` fields at all?

Reply:
That's why the `get/setProperty` method is there. And yes, spec and status were not expected to be exposed (at least, that was the idea).

Reply:
ok, understood. Given that we are only support get/set properties on the spec object, I am going to leave as it is . But I will add a comment to the code to explain that this can be extended to support set/get more properties on other objects.